### PR TITLE
Final merge of topic-gpio branch

### PR DESCRIPTION
**Pull request by pabigot**
Push `HEAD^` when it passes so the shippable override is not included.

### CODE REVIEW

**Review by mbolivar — APPROVED** [review]

**Review by mike-scott — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

File: drivers/gpio/gpio_sam.c [hunk 1]
```diff
@@ -16,78 +16,99 @@
...
-	} else {
-		pio->PIO_PUDR = mask;
-	}
 
 #if defined(CONFIG_SOC_SERIES_SAM4S) || \
 	defined(CONFIG_SOC_SERIES_SAME70) || \
 	defined(CONFIG_SOC_SERIES_SAMV71)
 	/* Setup Pull-down resistor. */
-	if ((flags & GPIO_PUD_MASK) == GPIO_PUD_PULL_DOWN) {
+	if (flags & GPIO_PULL_DOWN) {
```
Comment (mnkp) [inline drivers/gpio/gpio_sam.c]:
```
        } else if (flags & GPIO_PULL_DOWN) {
```

File: drivers/gpio/gpio_sam.c [hunk 2]
```diff
@@ -16,78 +16,99 @@
...
+	pio->PIO_PUDR = mask;
+#if defined(CONFIG_SOC_SERIES_SAM4S) || defined(CONFIG_SOC_SERIES_SAME70)
+	pio->PIO_PPDDR = mask;
+#endif
+	if (flags & GPIO_PULL_UP) {
 		/* Enable pull-up. */
 		pio->PIO_PUER = mask;
-	} else {
-		pio->PIO_PUDR = mask;
-	}
```
Comment (mnkp) [inline drivers/gpio/gpio_sam.c]: Let's remove this empty line. The code will read better. Originally it wasn't there.

**Review by mnkp — CHANGES_REQUESTED** [review]

**Review by ydamigos — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by vanwinkeljan — APPROVED** [review]

**Review by mnkp — APPROVED** [review]
Tested on efr32_slwstk6061a, nrf52840_pca10056, frdm_k64f, sam_e70_xplained, nucleo_wb55rg board with `tests/drivers/gpio/gpio_api_1pin/` test.

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
Tagging as DNM since it has the minions tweak commit

**Pull request merged**